### PR TITLE
docs(release): update release notes for readability

### PR DESCRIPTION
**Pull request by alevenberg**
- Add ordered lists so steps are easier to reference
- Move sections and add headers

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13020)

### CODE REVIEW

File: release/README.md [hunk 1]
```diff
@@ -58,21 +63,68 @@ tweak as needed.
...
+git push
+```
+
+**NOTE:** The Renovate bot will automatically update the Bazel deps in the
+quickstart `WORKSPACE.bazel` files after it sees the new release published.
+Watch for this PR to come through, kick off the builds, approve, and merge it.
+
+## 3. Creating the release
+
+**NOTE:** Do NOT create the release branch before the PR created in step 1 and 2
```
Comment (coryan) [inline release/README.md:119]: Only step 1 is blocking.  I think the ideal order is something like:

- Create the PR for CHANGELOG
- While that goes through review and builds: start the build for the version bump
- Merge the PR for the CHANGELOG
- Create the release
- Send the PR to bump the version number

**Review by coryan — CHANGES_REQUESTED** [review]

File: release/README.md (hunk 1 above)
Comment (alevenberg) [reply release/README.md:122]: Done

File: release/README.md [hunk 2]
```diff
@@ -10,31 +10,33 @@ project that need to create a new release. We expect the reader to be familiar
...
 enumerated below.
 
-### Verify CI passing
+### a. Verify CI passing
 
 Before beginning the release process, verify all CI builds are passing on the
 `main` branch. This is displayed in the GitHub page for the project.
 
-### Update the root CMakeLists.txt
+### b. Create a PR with to prepare the pre-release
```
Comment (coryan) [inline release/README.md:23]: s/with to/to/ ?

File: release/README.md [hunk 3]
```diff
@@ -58,21 +63,71 @@ tweak as needed.
...
 ci/cloudbuild/build.sh -t checkers-pr
 ```
 
-### Send a PR with all these changes
+#### Send a PR with all these changes
+
+```bash
+git add .
+git checkout -b release-changelog
+git commit -am "docs(release): update changelog for XX-20XX release" // Change this to the month of the release
```
Comment (coryan) [inline release/README.md:77]: s/XX-20XX release/$(date +%Y-%m) release/ and you can also `s;//.*;;`.  In any case `# ` is the comment character for bash.

File: release/README.md [hunk 4]
```diff
@@ -58,21 +63,71 @@ tweak as needed.
...
+
+### c. Send the PR for review
+
+**NOTE:** Do NOT send this PR for review before the release is created in step
+3\.
+
+```bash
+git add .
+git checkout -b bump-rc
+git commit -am "chore: version bump to X.XY.Zrc" // Change this to the the correct version
```
Comment (coryan) [inline release/README.md]: s/X.XY.Zrc/X.XY.Z-rc/

It may be possible to get the version using:

```
$(sed -n 's/.* VERSION //p' CMakeLists.txt)
```

**Review by coryan — APPROVED** [review]
A few (maybe) improvements.

File: release/README.md (hunk 3 above)
Comment (alevenberg) [reply release/README.md]: Done.

I don't know what you mean by `s;//.*;;.`.

File: release/README.md (hunk 2 above)
Comment (alevenberg) [reply release/README.md]: Done

File: release/README.md (hunk 3 above)
Comment (coryan) [reply release/README.md]: That you could remove the comment... which you did.

**Pull request merged**